ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiInMemoryWorldState.java

```diff
@@ -41,12 +41,16 @@ public Hash rootHash() {
   @Override
   public void persist(final BlockHeader blockHeader) {
     final BonsaiWorldStateUpdater localUpdater = updater.copy();
+    final BonsaiWorldStateKeyValueStorage.Updater stateUpdater = worldStateStorage.updater();
     try {
       final Hash newWorldStateRootHash = rootHash();
-      prepareTrieLog(blockHeader, localUpdater, newWorldStateRootHash);
+      final TrieLogLayer trieLogLayer =
+          prepareTrieLog(blockHeader, localUpdater, newWorldStateRootHash);
+      persistTrieLog(blockHeader, newWorldStateRootHash, trieLogLayer, stateUpdater);
       worldStateBlockHash = blockHeader.getHash();
       worldStateRootHash = newWorldStateRootHash;
     } finally {
+      stateUpdater.commit();
```

Comment:
Should this be up at line 47/50 instead?  Do we want to commit the state updater if persistTrieLog throws an error?

Reply:
Yes you right. Good catch

Reply:
done thanks